package-system/AWSGameLiftServerSDK/build-linux.sh

```diff
@@ -84,133 +117,103 @@ then
 
     DOCKER_INPUT_ARCHITECTURE=amd64
     TARGET_DOCKER_PLATFORM_ARG=linux/amd64
-    DOCKER_BUILD_ARG=1
 
 elif [ "${TARGET_ARCH}" = "aarch64" ] 
 then
     echo "Processing Docker for aarch64"
 
     DOCKER_INPUT_ARCHITECTURE=arm64v8
     TARGET_DOCKER_PLATFORM_ARG=linux/arm64/v8
-    DOCKER_BUILD_ARG=3
+
 else
     echo "Unsupported architecture ${TARGET_ARCH}"
     exit 1
 fi
 
-# Prepare to build on both Ubuntu 20.04 and Ubuntu 22.04 based docker images
-
+# 
+# Prepare the docker base context based on ${TEMP_FOLDER} 
 mkdir -p ${TEMP_FOLDER}
 cp -f ${DOCKER_BUILD_SCRIPT} ${TEMP_FOLDER}/
 
-# Args
-# $1 : Ubuntu version
-# $2 : Include
-# $3 : Docker run platform
-
-function execute_docker() {
-
-    # Determine the openssl version based on the ubuntu version (20.04/OpenSSL 1.1.1.x vs 22.04/OpenSSL 3.x)
-    if [ $1 = "20.04" ]
-    then
-        echo "Preparing for OpenSSL 1.1.1.x version"
-        BIN_SUBFOLDER_NAME=openssl-1
-    elif [ $1 = "22.04" ]
-    then
-        echo "Preparing for OpenSSL 3.x version"
-        BIN_SUBFOLDER_NAME=openssl-3
-    else
-        echo "Unsupported base build image ubuntu version ${1}"
-        exit 1
-    fi
+echo "Building on ubuntu public.ecr.aws/ubuntu/ubuntu:${UBUNTU_BASE}"
+
+# Build the Docker Image 
+DOCKER_IMAGE_NAME=${DOCKER_IMAGE_NAME_BASE}_${DOCKER_INPUT_ARCHITECTURE}_3p
+echo DOCKER_IMAGE_NAME=${DOCKER_IMAGE_NAME}
+
+echo "Building the docker build script for ${DOCKER_IMAGE_NAME_BASE} on ${DOCKER_INPUT_ARCHITECTURE} for Ubuntu $1"
+echo ""
+echo docker build --build-arg INPUT_DOCKER_BUILD_SCRIPT=${DOCKER_BUILD_SCRIPT} \
+             --build-arg INPUT_ARCHITECTURE=${DOCKER_INPUT_ARCHITECTURE} \
+             --build-arg INPUT_IMAGE=ubuntu:${UBUNTU_BASE} \
+             --build-arg INPUT_DEPENDENT_PACKAGE_FOLDERS=${DEP_PACKAGES_DOCKER_FOLDERNAMES} \
+             -f Dockerfile -t ${DOCKER_IMAGE_NAME}:latest temp 
+docker build --build-arg INPUT_DOCKER_BUILD_SCRIPT=${DOCKER_BUILD_SCRIPT} \
+             --build-arg INPUT_ARCHITECTURE=${DOCKER_INPUT_ARCHITECTURE} \
+             --build-arg INPUT_IMAGE=ubuntu:${UBUNTU_BASE} \
+             --build-arg INPUT_DEPENDENT_PACKAGE_FOLDERS=${DEP_PACKAGES_DOCKER_FOLDERNAMES} \
+             -f Dockerfile -t ${DOCKER_IMAGE_NAME}:latest temp 
```

Comment:
Can't we use set -x instead to have bash output the command that is running without needing to copy and paste in echo?
https://stackoverflow.com/a/5750463
```suggestion
set -o xtrace
docker build --build-arg INPUT_DOCKER_BUILD_SCRIPT=${DOCKER_BUILD_SCRIPT} \
             --build-arg INPUT_ARCHITECTURE=${DOCKER_INPUT_ARCHITECTURE} \
             --build-arg INPUT_IMAGE=ubuntu:${UBUNTU_BASE} \
             --build-arg INPUT_DEPENDENT_PACKAGE_FOLDERS=${DEP_PACKAGES_DOCKER_FOLDERNAMES} \
             -f Dockerfile -t ${DOCKER_IMAGE_NAME}:latest temp 
set +o xtrace
```

Reply:
thats very helpful! I'll apply that across the board

Reply:
This approach eats up the exit code with the second trace, will use a different approach (mentioned below)